content/blog/which-unique-identifier-is-right-for-you.md

```diff
@@ -30,75 +29,73 @@ Let's break down UUID for better visibility.
 
-It uses a UNIX Timestamp in milliseconds and a Cryptographically secure source of randomness to generate 2nd part of ULID (Randomness).
+It uses a UNIX timestamp in milliseconds and a cryptographically secure source of randomness to generate the second part of the ULID (randomness).
 
 #### Pros
 
-- Lexicographical Sorting is the biggest highlight of ULID. It can be sorted by their natural order.
-- It's more readable and URL friendly and moreover, it is compact in length compared to UUID. Hence it saves space in terms of storage.
-- ULID is the most feasible solution if there is a need for sequentiality or order.
+- Lexicographical sorting is the biggest highlight of ULID, allowing IDs to be sorted in their natural order.
+- More readable and compact than UUID, saving storage space.
+- Ideal solution if there is a need for sequentiality or order.
 
 #### Cons
 
-- Randomness is limited in timestamp, It is possible to generate multiple IDs within the same millisecond.
-- Due to limited randomness, It doesn't guarantee a collision-resistant solution.
-- It has yet to be adopted much due to compatibility issues with UUID. (if systems are already using UUID).
+- Randomness is limited in the timestamp, so it is possible to generate multiple IDs within the same millisecond.
+- Due to limited randomness, it doesn't guarantee a collision-resistant solution.
+- Not widely adopted due to compatibility issues with UUID systems.
 
 ---
 
-### CUID
+### CUID (Collision-Resistant Unique Identifier)
 
-Collision-resistant Unique Identifier also known as CUID, It's secure, guarantees strong collision-resistant and provides good performance.
+CUIDs are designed for high collision resistance and performance, with a default length of 24 characters but configurable up to 32 characters.
 
 To react 50% chance of collision, you'd need to generate roughly 4,000,000,000,000,000,000 IDs. Which is quite huge.
 
-It is configurable to at max 32 characters long, the default length is 24 characters.
-
-Let's break it down to understand it better
-
 {{< figure src="/images/2024/which-unique-identifier-is-right-for-you/CUID-architecture.png" >}}
 
-It uses a combination of UNIX Time in Millisecond + Salt + Session Count + Fingerprint and Hash these values with SHA3 Hashing Algorithm prefix by random alphabet.
+It uses a combination of UNIX time in milliseconds, salt, session count, fingerprint, and hashes these values with the SHA3 hashing algorithm, prefixed by a random alphabet.
 
 #### Pros
 
-- The main advantage of CUID is Collision-Resistant, It's quite strong in ensuring uniqueness. It focuses on keeping entropy as high as possible to ensure correct randomness.
-- It's secure and non-guessable, provides offline ID generation support, URL friendly (Doesn't contain special symbols).
-- CUID is horizontally scalable, it can generate IDs on various machines with coordination.
+- Strong collision resistance, making it highly unique.
```

Comment:
I would add
"**It focuses on keeping entropy as high as possible to ensure correct randomness.**"

Because it's a key highlight provided by CUID Docs. I read it thoroughly